src/main/groovy/edu/oregonstate/mist/personsapi/PersonsResource.groovy

```diff
@@ -987,10 +987,16 @@ class PersonsResource extends Resource {
             [true, "phoneType", 2,
              { String phoneType -> bannerPersonsReadDAO.isValidPhoneType(phoneType) }
             ],
-            [false, "areaCode", 3, null],
-            [false, "phoneNumber", 7, null],
-            [false, "phoneExtension", 4, null],
-            [false, "primaryIndicator", 5, null],
+            [true, "areaCode", 3, 
+             { String areaCode -> areaCode =~ /^[0-9]{1,3}$/ }
+            ],
+            [true, "phoneNumber", 7,
+             { String phoneNumber -> phoneNumber =~ /^[0-9]{1,7}$/ }
+            ],
+            [false, "phoneExtension", 4,
+             { String phoneExtension -> phoneExtension =~ /^[0-9]{0,4}$/ }
```

Comment:
Thats a good point. It ended up not making a difference because empty string is falsey and it ends up being null in the database. I still changed it to `{1,4}` because it makes more sense to me.